File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPInvisible.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPInvisible implements PlayerStateProperty<Boolean> {
    @Override
    public String getConfigName() {
        return "invisible";
    }

    @Override
    public Boolean getDefaultSerializedProperty() {
        return false;
    }

    @Override
    public Boolean getSerializedPropertyFromPlayer(Player player) {
        return player.isInvisible();
    }

    @Override
    public void restoreProperty(Player player, Boolean invisible) {
        player.setInvisible(invisible);
    }

    @Override
    public Class<Boolean> getTypeClass() {
        return Boolean.class;
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPInvisible.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPInvisible extends PlayerStateProperty[Boolean]:
  override def getConfigName: String = "invisible"

  override def getDefaultSerializedProperty: Boolean = false

  override def getSerializedPropertyFromPlayer(player: Player): Boolean =
    player.isInvisible

  override def restoreProperty(player: Player, invisible: Boolean): Unit =
    player.setInvisible(invisible)